### ISSUE TITLE
Tracing of jstack invocation triggers NPE

### ISSUE THREAD

**Issue by Stephan202** [DataDog/dd-trace-java#5838]
When we execute `jstack 1` the command succeeds (a thread dump is written to stdout), but the following is logged to stderr:
```
Picked up JAVA_TOOL_OPTIONS:  -javaagent:/datadog-lib/dd-java-agent.jar
OpenJDK 64-Bit Server VM warning: Sharing is only supported for boot loader classes because bootstrap classpath has been appended
[dd.trace 2023-09-05 06:33:03:267 +0000] [main] ERROR datadog.trace.bootstrap.Agent - Throwable thrown while installing the Datadog Tracer
[dd.trace 2023-09-05 06:33:03:271 +0000] [Finalizer] ERROR datadog.trace.agent.core.CoreTracer - Error while finalizing DDTracer.
java.lang.reflect.InvocationTargetException
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at datadog.trace.bootstrap.Agent.installDatadogTracer(Agent.java:566)
	at datadog.trace.bootstrap.Agent.access$400(Agent.java:67)
	at datadog.trace.bootstrap.Agent$InstallDatadogTracerCallback.execute(Agent.java:458)
	at datadog.trace.bootstrap.Agent.start(Agent.java:293)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at datadog.trace.bootstrap.AgentBootstrap.agentmain(AgentBootstrap.java:73)
	at datadog.trace.bootstrap.AgentBootstrap.premain(AgentBootstrap.java:54)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndStartAgent(InstrumentationImpl.java:491)
	at java.instrument/sun.instrument.InstrumentationImpl.loadClassAndCallPremain(InstrumentationImpl.java:503)
Caused by: java.lang.NoClassDefFoundError: sun/misc/Unsafe
	at org.jctools.util.UnsafeAccess.getUnsafe(UnsafeAccess.java:54)
	at org.jctools.util.UnsafeAccess.<clinit>(UnsafeAccess.java:44)
	at org.jctools.counters.CountersFactory.createFixedSizeStripedCounter(CountersFactory.java:14)
	at datadog.trace.agent.core.monitor.TracerHealthMetrics.<init>(TracerHealthMetrics.java:38)
	at datadog.trace.agent.core.monitor.TracerHealthMetrics.<init>(TracerHealthMetrics.java:128)
	at datadog.trace.agent.core.CoreTracer.<init>(CoreTracer.java:540)
	at datadog.trace.agent.core.CoreTracer.<init>(CoreTracer.java:117)
	at datadog.trace.agent.core.CoreTracer$CoreTracerBuilder.build(CoreTracer.java:432)
	at datadog.trace.agent.tooling.TracerInstaller.installGlobalTracer(TracerInstaller.java:25)
	... 20 more
Caused by: java.lang.ClassNotFoundException: sun.misc.Unsafe
	at datadog.trace.bootstrap.DatadogClassLoader.loadClassBytes(DatadogClassLoader.java:144)
	at datadog.trace.bootstrap.DatadogClassLoader.findClass(DatadogClassLoader.java:115)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:587)
	at datadog.trace.bootstrap.DatadogClassLoader.loadClass(DatadogClassLoader.java:109)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:520)
	... 29 more
java.lang.NullPointerException: Cannot invoke "java.lang.Thread.run()" because "this.shutdownCallback" is null
	at datadog.trace.agent.core.CoreTracer.finalize(CoreTracer.java:668)
	at java.base/java.lang.System$2.invokeFinalize(System.java:2297)
	at java.base/java.lang.ref.Finalizer.runFinalizer(Finalizer.java:88)
	at java.base/java.lang.ref.Finalizer$FinalizerThread.run(Finalizer.java:173)
```

While this doesn't seem to have a functional impact, it does confuse users. Perhaps it can be avoided? :)

**Comment by Stephan202** [comment]
Running `JAVA_TOOL_OPTIONS="" jstack 1` avoids the issue. In our case `JAVA_TOOL_OPTIONS` is inherited from the Kubernetes environment. (I assume from the Datadog Cluster Agent or Datadog Admission Controller.)

**Comment by Stephan202** [comment]
The same (again, benign) stack trace is also logged for e.g. `jcmd 1 JFR.check` and `jcmd 1 JFR.start`.

**Comment by mcculls** [comment]
Thanks for reporting this @Stephan202, we'll remove that stack trace in the next release.

As you saw it shouldn't affect the actual command, but it clutters up the console.

Unsetting `JAVA_TOOL_OPTIONS` before running serviceability commands is the recommended workaround.

**Comment by mcculls** [comment]
Hi @Stephan202 - this snapshot of the Java tracer should resolve the issue you saw:

https://oss.sonatype.org/content/repositories/snapshots/com/datadoghq/dd-java-agent/1.21.0-SNAPSHOT/dd-java-agent-1.21.0-20230908.211604-39.jar

the fix will be part of the next release, 1.21.0

**Comment by Stephan202** [comment]
Nice; thanks for the quick fix :rocket:!
